asserts/ifacedecls.go

```diff
@@ -241,20 +241,129 @@ func (c *AttributeConstraints) Check(attrs map[string]interface{}) error {
 	return c.matcher.match("", attrs)
 }
 
-func checkMapOrShortcut(context string, v interface{}) (m map[string]interface{}, outcomeShortcut string, err error) {
+// rules
+
+var (
+	validSnapType  = regexp.MustCompile("^(?:core|kernel|gadget|app)$")
+	validSnapID    = regexp.MustCompile("^[a-z0-9A-Z]{32}$")                                             // snap-ids look like this
+	validPublisher = regexp.MustCompile("^(?:[a-z0-9A-Z]{32}|[-a-z0-9]{2,28}|\\$[a-z](?:-?[a-z0-9])*)$") // account ids look like snap-ids or are nice identifiers, support our own special markers $MARKER
```

Comment:
Isn't the middle one included in thelast one, except for names beginning with digits and dashes?